modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/OnflightMessageTracker.java

```diff
@@ -878,4 +906,70 @@ public void dumpMessageStatusToFile(File fileToWrite) throws AndesException {
         }
     }
 
+    /**
+     * Concurrent List like implementation.
+     * This implementation will create thread safe LinkedList.
+     * Two threads won't be able to execute same operation simultaneously
+     * with this LinkedList.
+     *
+     * @param <T> Type of list items
+     *
+     */
+    private static class ConcurrentTrackingList<T> {
```

Comment:
Oh yes, We had a  discussion on this. Sorry, I forgot the reason behind using this class. Let's keep it as it is. Thanks for the explanation. :)